store/tikv/txn.go

```diff
@@ -319,34 +320,28 @@ func (txn *tikvTxn) close() {
 }
 
 func (txn *tikvTxn) Rollback() error {
+	if !txn.valid {
+		return kv.ErrInvalidTxn
+	}
 	// Clean up pessimistic lock.
 	if txn.IsPessimistic() && txn.committer != nil {
-		err := txn.rollbackPessimisticLock()
+		err := txn.rollbackPessimisticLocks()
```

Comment:
Is it possible rollback a non-pessimistic lock in the pessimistic transaction?

Reply:
There is a `txn.IsPessimistic()` @youjiali1995

Reply:
Sorry, I didn't make myself clear. Is it possible the pessimistic transaction holds some non-pessimistic locks here?

Reply:
If `Rollback` is called, `Commit` will not be called, non-pessimistic keys is not written.